### PR TITLE
Implement arithmetic const nodes.

### LINKED ISSUE #195: Implement arithmetic resource

### PR DESCRIPTION
**Pull request by cqc-alec**
Closes #195 .

### CODE REVIEW

File: src/extensions/arithmetic/float_types.rs [hunk 1]
```diff
@@ -13,14 +14,45 @@ pub const RESOURCE_ID: SmolStr = SmolStr::new_inline("arithmetic.float.types");
 /// Identfier for the 64-bit IEEE 754-2019 floating-point type.
 const FLOAT_TYPE_ID: SmolStr = SmolStr::new_inline("float64");
 
+fn float64_custom_type() -> CustomType {
```
Comment (ss2165) [inline src/extensions/arithmetic/float_types.rs:17]: can this be a const? (Using `CustomType::new_simple`)

File: src/extensions/arithmetic/float_types.rs [hunk 2]
```diff
@@ -13,14 +14,45 @@ pub const RESOURCE_ID: SmolStr = SmolStr::new_inline("arithmetic.float.types");
...
+
+    fn check_custom_type(&self, typ: &CustomType) -> Result<(), CustomCheckFailure> {
+        if typ.clone() == float64_custom_type() {
+            Ok(())
+        } else {
+            Err(CustomCheckFailure::Message(
+                "Floating-point constant type mismatch.".into(),
+            ))
+        }
+    }
```
Comment (ss2165) [inline src/extensions/arithmetic/float_types.rs:51]: I think float can implement `crate::values::KnownTypeConst` then just use `self.check_known_type(typ)` here?

**Review by ss2165 — CHANGES_REQUESTED** [review]
I think the float implementation can be simplified by copying the `ConstUsize`

File: src/extensions/arithmetic/float_types.rs [hunk 3]
```diff
@@ -13,14 +14,42 @@ pub const RESOURCE_ID: SmolStr = SmolStr::new_inline("arithmetic.float.types");
 /// Identfier for the 64-bit IEEE 754-2019 floating-point type.
 const FLOAT_TYPE_ID: SmolStr = SmolStr::new_inline("float64");
 
+const FLOAT64_CUSTOM_TYPE: CustomType =
+    CustomType::new_simple(FLOAT_TYPE_ID, RESOURCE_ID, TypeBound::Copyable);
+
 /// 64-bit IEEE 754-2019 floating-point type
 pub fn float64_type() -> Type {
```
Comment (ss2165) [inline src/extensions/arithmetic/float_types.rs:21]:
```suggestion
pub const FLOAT64_TYPE: Type = Type::new_extension(FLOAT64_CUSTOM_TYPE);
```

**Review by ss2165 — COMMENTED** [review]
Can be even more const

File: src/extensions/arithmetic/float_types.rs (hunk 3 above)
Comment (cqc-alec) [reply src/extensions/arithmetic/float_types.rs]: Yup, done.

File: src/extensions/arithmetic/conversions.rs [hunk 4]
```diff
@@ -24,7 +24,7 @@ fn ftoi_sig(arg_values: &[TypeArg]) -> Result<(TypeRow, TypeRow, ResourceSet), S
     let [arg] = collect_array(arg_values);
     let n: u8 = get_width(arg)?;
     Ok((
-        vec![float64_type()].into(),
+        vec![FLOAT64_TYPE].into(),
```
Comment (ss2165) [inline src/extensions/arithmetic/conversions.rs:27]: now I'm just getting annoying but...
```suggestion
        type_row![FLOAT64_TYPE],
```

**Review by ss2165 — APPROVED** [review]

File: src/extensions/arithmetic/conversions.rs (hunk 4 above)
Comment (cqc-alec) [reply src/extensions/arithmetic/conversions.rs]: Thanks, lots of instances of that. Replaced.

**Pull request merged**